Hotfix ns2b3 viz

JenkeScheen opened the pull request:
Adds a fitness view of ZIKV NS2B3 so that we can get a demo for P1/P2 for `argos`. This uses DMS data from the Evans lab, analyzed by the Bloom lab (see [this notebook](https://github.com/jbloomlab/ZIKV_DMS_with_EvansLab/blob/master/analysis_notebook.ipynb), and specifically [this csv](https://github.com/jbloomlab/ZIKV_DMS_with_EvansLab/blob/master/results/muteffects/unscaled_muteffects.csv)). This PR enables usage of 
```
pose_to_viz.py --viz-target ZIKV-NS2B-NS3pro --pose ZIKV_NS2B3_ligands.sdf --protein ZIKV_NS2B3_apo.pdb --out view.html --color-method fitness
```
It also fixes an oversight: when running this script we were not aligning input structures to the master structures found in `metadata` (this was only happening in the end2end docking workflow in the `prep` phase). Now, users can add `align=True` to `HTMLVisualizer()` to enable this. 

- [x] This PR is pending Jesse Bloom's approval of the set threshold of -1.0.


## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

hmacdope commented on asapdiscovery-data/asapdiscovery/data/fitness.py:
```diff
@@ -64,14 +67,19 @@ def apply_bloom_abstraction(fitness_dataframe: pd.DataFrame) -> dict:
             total count (~confidence)
         ]
     """
+    # add this column in case we're pulling in an experiment that has different data. We need to find
```
We should pre-process so that it is in a consistent format, perhaps the submitting labs may not have a common language, but we should to avoid an explosion of complexity or inconsistent data that looks consistent.

hmacdope replied:
Raised issue #649

JenkeScheen replied:
agreed - imo fine for now, let's revisit the next time we receive a divergent format so that we can generalize between the three.

hmacdope commented on asapdiscovery-data/asapdiscovery/data/fitness.py:
```diff
@@ -140,35 +148,48 @@ def parse_fitness_json(target: TargetTags) -> pd.DataFrame:
+        elif target == "SARS-CoV-2-Mpro":
+            # simpler; can just query the correct gene in the JSON.
+            fitness_scores_bloom = fitness_scores_bloom[
+                fitness_scores_bloom["gene"] == "nsp5 (Mpro)"
```
Can this use `_TARGET_TO_GENE`?

JenkeScheen replied:
yep, just tested and works fine.

hmacdope commented on asapdiscovery-data/asapdiscovery/data/fitness.py:
```diff
@@ -140,35 +148,48 @@ def parse_fitness_json(target: TargetTags) -> pd.DataFrame:
-        ] -= 208  # reindex to set residue numbers to correct values
-    elif target == "SARS-CoV-2-Mpro":
-        # simpler; can just query the correct gene in the JSON.
+    ## START TODO
```
We can abstract this, but can push down road for now.

JenkeScheen replied:
agreed, we can probably loop this into https://github.com/choderalab/asapdiscovery/issues/649

hmacdope reviewed (approved):
Few small comments, but we can push down the road as you like.

hmacdope commented on asapdiscovery-data/asapdiscovery/data/fitness.py:
```diff
@@ -140,35 +148,48 @@ def parse_fitness_json(target: TargetTags) -> pd.DataFrame:
+            ]
+    # for ZIKV NS2B3 it's simpler - no need to subselect the target from the genome.
+    elif target == "ZIKV-NS2B-NS3pro":
+        threshold = 0.0
```
Current thresh 0.0 for ZIKV?

hmacdope commented on asapdiscovery-data/asapdiscovery/data/fitness.py:
```diff
@@ -140,35 +148,48 @@ def parse_fitness_json(target: TargetTags) -> pd.DataFrame:
+            ]
+    # for ZIKV NS2B3 it's simpler - no need to subselect the target from the genome.
+    elif target == "ZIKV-NS2B-NS3pro":
+        threshold = 0.0
```
Current thresh 0.0 for ZIKV?

JenkeScheen replied:
yep, it's a completely different experiment so the dynamic range is different. For this DMS analysis, values are a log transformation from 0: a value of -1 means 10x less fit than wildtype, a value of 2 means 100x more fit than mutant.

I'm double-checking with Jesse Bloom to make sure I took the correct score here but I'm fairly confident, so we can merge imo

hmacdope replied:
issue raised #654

hmacdope commented:
I have done the abstraction of that function. Would you mind checking to see if the zika example works? Is the structure available for public consumption?

hmacdope commented:
Hang on something was changed in `dab52ee` to make the values different, let me check

hmacdope commented:
> Hang on something was changed in `dab52ee` to make the values different, let me check

Ignore me I am stupid we changed the color scheme.

JenkeScheen commented:
> Ignore me I am stupid we changed the color scheme.

Oh right, I removed some colors from that list. Now instead of white -> slightly pink -> slightly red -> red -> very red, we color as white -> red -> very red -> super red etc. Makes it much easier to distinguish between 0 fit mutants and 1 fit mutant. 

We should still overlay a color bar/ legend to the interactive view.

hmacdope reviewed (approved):
I think this is good to go for now, pending followup with JB for fitness thresh.
